### ISSUE TITLE
Format `type: ignore` comments

### ISSUE THREAD

**Issue by KotlinIsland** [psf/black#2601]
Type ignore are standard constructs in python code. I think it would be based if black formatted them

```py
a: SUS = "AMONGUS" + 1 # type:ignore     [name-defined      ,    operator     ,misc    ]
```
Could be formatted to:
```py
a: SUS = "AMONGUS" + 1 # type: ignore[name-defined, operator, misc]
```

**Comment by isidentical** [comment]
Just a side note: beside simply putting a single whitespace after the separator (since trailing white-space is ignored by the python tokenizer, `#type: ignore` and `#type:ignore` is basically the same thing in the AST), anything else would change the AST.

**Comment by JelleZijlstra** [comment]
Yes, everything after the ignore has only implementation-defined meaning. I suppose we could format assuming mypy semantics, though.

**Comment by KotlinIsland** [comment]
PyCharm assumes mypy semantics 
![image](https://user-images.githubusercontent.com/65446343/141861185-dd037ec2-5ee0-4cfd-a93d-99f4bcf5fd33.png)

**Comment by cobaltt7** [comment]
Part of this is implemented in #4645 (Excess spaces after the `#` and `:` are removed, but the value is left unchanged). IMO, the value shouldn't be changed for the reasons mentioned here: there's only implementation-defined meaning and we would have to assume mypy semantics, and the AST safety check would have to be modified for this case. Please share your thoughts.
